### ISSUE TITLE
Original search query is still ran

### ISSUE THREAD

**Issue by Floddy** [algolia/algoliasearch-magento#213]
When going to the fullpage search results, the Magento search query is still ran, then the result page is replaced using js.

Since the query is ran, this uses unnecessary resources. Also, since the whole content is removed, other blocks apart from the search result is also removed.

A more proper implementation would probably be making a rewrite for the catalogsearch page, having its own layout or layout updates, and perhaps, as a bonus, splitting up the Algolia template in two: one for topsearch, one for fullpage.

**Comment by maxiloc** [comment]
@Floddy Yes we ran the query for seo reason and we do not want to disable that by default. Maybe we can add an option to disable this query.

For me it does not make sense to make a php rewrite of the magento page. The extension is designed to do instant search via js (because it's way faster) rather than via the magento backend.

What do you think ?

cc @redox

**Comment by maxiloc** [comment]
Fixed with 09f23b8.

@Floddy you will now have the option to disable that in the advanced section of the configuration. I would recommend to let it enable if you want search engines like google to be able to crawl search pages.

**Comment by Floddy** [comment]
I may have missed something here - I thought it was the original Magento Mysql fulltext that was still ran - not through algolia - since I got somewhat different results on an installation. If it's all ran through Algolia, I see the point.

Though there is still the issue of removing the whole content in .main with js, since every other block is removed too. I'd like if the page setup was intact (2-col-left), it you removed only the search results and the filters in the sidebar, and replaced them with the algolia results and the facets - but I suspect this could be tricky since Algolia inserts everything 'in one chunk' and therefore can't insert the facects in the sidebar and the results in the main content?

The reason I looked into this is that we use a recommendation engine, and we usually have recommendations on the search result page (other customers that searched for this... etc), but these are being removed with the js now. We can probably come up with a custom solution, but it would be nice if it could work out of the box in some way.
